pkg/db/client.go

```diff
@@ -445,36 +431,40 @@ func (c *ClientSqlx) BeginTx(ctx context.Context, ops *sql.TxOptions) (*sqlx.Tx,
 	return res.(*sqlx.Tx), err
 }
 
-func (c *ClientSqlx) WithTx(ctx context.Context, ops *sql.TxOptions, do func(ctx context.Context, tx *sqlx.Tx) error) (err error) {
-	var tx *sqlx.Tx
-	tx, err = c.BeginTx(ctx, ops)
-	if err != nil {
-		return err
-	}
-
-	defer func() {
+func (c *ClientSqlx) WithTx(ctx context.Context, ops *sql.TxOptions, do func(ctx context.Context, tx *sqlx.Tx) error) error {
+	_, err := c.executor.Execute(ctx, func(ctx context.Context) (res any, err error) {
+		var tx *sqlx.Tx
+		tx, err = c.db.BeginTxx(ctx, ops)
 		if err != nil {
-			errRollback := tx.Rollback()
-			if errRollback != nil {
-				err = multierror.Append(err, fmt.Errorf("can not rollback tx: %w", errRollback))
+			return nil, err
+		}
 
-				return
+		defer func() {
+			if err != nil {
```

Comment:
What happens if err is nil at this place because `do` did panic? I would expect the function to catch the panic and return an error for it, right?